blueprint/core/blueprint_ShadowView.h

```diff
@@ -89,6 +91,54 @@ namespace blueprint
         }
     };
 
+    template <typename Setter, typename ...Args>
+    const auto getYogaNodeFloatSetter(Setter setter, Args... args) {
+      return [=](const juce::var& value, YGNodeRef node) {
+        if(value.isDouble()) {
```

Comment:
Do you think we should throw here?

```
if (!value.isDouble()) {
  throw std::invalid_argument("Layout properties must have numeric values.")
}
setter(node....);
return true;
```
That way we can catch this and bubble the error into js?

Reply:
@nick-thompson I'm not actually sure.  If we did simply the above change, we'd break percentage based values (since we'd throw before they got tried).  If we attempt to parse % first, similarly we'd never call the base setter with a `YGValue{0.0f, YGUndefined}`, which is what currently happens if your value isn't a double or a string with `%`.  This might be fine, but I basically just converted the previous macros to something maximally template-based and logically equivalent.

I'm down with throwing on invalid values, but it might take a bit of thought (e.g. also we'd have to decide when to throw in the % case, since the existing call to juce::String::getFloatValue won't throw).

Reply:
Ahh ok, good point. Let's keep it as is then!

We can always revisit later if we find an elegant way to do it, but I think falling back to "ignore" behavior is also ok, since that's what we see in web.